crates/swc_css_parser/tests/fixture.rs

```diff
@@ -20,7 +20,14 @@ impl Visit for AssertValid {
             }
 
             match &tokens.tokens[0].token {
```

Comment:
This seems like a regression?

Reply:
Yep, because we should not try to test it in this way, it is wrong, tokens from custom properties and pseudo selectors can't be reparsed in the same, because they are in other context, i.e. we have
```
:pseudo(foo { color: red }) { }
```

When we try to parse it using tokens input we send tokens to parser:
```
foo { color: red }
```

and our parser thinks it is `QualifiedRule`, but it is wrong, because it is just tokens of pseudo selectors, in `any-value` or declaration-value` according spec we can have almost any value, but they can't be reparsing in right way.

I thought to raise this issue and fix our tests a little later.

Reply:
Ideally we should remove this check and output tokens as is in our tests, I can fix it here, it is easy

Reply:
You can compare real characters in code and our tokens, and they are same

Reply:
Removed and nothing changes :smile:

Reply:
Removing visitor is even more regression, because the only purpose of the visitor is to `panic` if input is invalid. It's not normalizer.

Reply:
But we can't use `parse_tokens` in this case, because `parse_tokens` expecting we will have `Stylesheet` at the top, but here no `Stylesheet`, it is part of custom property or pseudo selector.

I think we need to use other approach to tests tokens, we should parse original code (in fixtures) to AST, then use `parse_tokens` to parse original code,and then compare AST, so we can check AST is the same as for tokens input and as for string input

Reply:
Also we can add just `parse_tokens` tests and snapshot them, so we can see what is changed in our tokens

Reply:
You can try to remove filtering tokens in this branch and you will see why we can't test tokens in this way...